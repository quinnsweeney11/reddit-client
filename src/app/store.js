import { configureStore } from "@reduxjs/toolkit";
import counterReducer from "../features/counter/counterSlice";
import postContainerReducer from "../containers/postContainer/postContainerSlice";
import filterReducer from "../features/filter/filterSlice";
import searchReducer from "../features/search/searchSlice";

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    postContainer: postContainerReducer,
    filter: filterReducer,
    search: searchReducer,
  },
});
