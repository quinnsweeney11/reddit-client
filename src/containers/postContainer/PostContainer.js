import React from 'react';
import { useSelector } from 'react-redux';

import { Post } from '../../features/post/Post';
import './postContainer.css'

import ReactLoading from 'react-loading';

export function PostContainer (props) {
    // const dispatch = useDispatch();
    const { isLoading, hasError, posts, showPosts } = useSelector((state) => state.postContainer)
    const currentFilter = useSelector((state) => state.filter);

    if(isLoading){
        return(
            <div className="nonpost-container">
                <ReactLoading className="loader" type='bars' color='#519fcf' height={'15%'} width={'10%'} />
            </div>
        )
    } else if(hasError){
        return(
            <div className="nonpost-container">
                <h1>Error loading posts, please try again later :(</h1>
            </div>
        )
    }

    if(currentFilter.currentFilter === 'new' && !showPosts){
        return (
            <div className="nonpost-container">
                <Post newWarn={true} />
            </div>
        )
    }

    return (
        <div className="PostContainer">
            {posts.map((post, i) => {
                return <Post key={i} data={posts[i].data}/>
            })}
        </div>
    )
}