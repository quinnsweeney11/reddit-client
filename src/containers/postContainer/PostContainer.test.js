import React from "react";
import { PostContainer } from "./PostContainer";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import { store } from "../../app/store";

describe("snapshot", () => {
  it("main", () => {
    const component = renderer.create(
      <Provider store={store}>
        <PostContainer />
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
