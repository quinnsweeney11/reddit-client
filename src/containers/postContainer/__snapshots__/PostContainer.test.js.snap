// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot main 1`] = `
<div
  className="PostContainer"
>
  <div
    className="search"
  >
    <form
      class="search-form"
      onSubmit={[Function]}
    >
      <input
        class="search-text"
        onChange={[Function]}
        placeholder="subreddit"
        type="text"
        value="r/"
      />
      <input
        class="search-button"
        onClick={[Function]}
        type="button"
        value="Search"
      />
    </form>
  </div>
</div>
`;
